### ISSUE TITLE
Freezing problem

### ISSUE THREAD

**Issue by SeonAntti** [OpenRCT2/OpenRCT2#20607]
### Operating System

Windows 11

### OpenRCT2 build

OpenRCT2, v0.4.5-202-g0a3573e (0a3573e on develop) provided by GitHub

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

This issue has to do with park objects and/or asset packs

### Describe the issue

The game crashes every time I try to name the specific rollercoaster.

### Steps to reproduce

The game crashes when I click the name tag symbol of this rollercoaster.

### Attachments

![Bug report](https://github.com/OpenRCT2/OpenRCT2/assets/139607405/a3175aff-dff4-4c6d-8195-fd0846c2dc3b)

Here is a screenshot

**Comment by ZehMatt** [comment]
Can you provide a save file, are you also using any plugins?

**Comment by SeonAntti** [comment]
I had an adblocker on but turning it off did not fix it. Unfortunately I am not very computer savvy but I will try to provide a save file. When I drop or paste the file here, I get "We don't support that file type" message. I can ask later on if my more computer savvy friend can assist me.

**Comment by James103** [comment]
Will you please zip the save file you're trying to post, then post the resulting zip file?

**Comment by SeonAntti** [comment]
[Botany Breakers.zip](https://github.com/OpenRCT2/OpenRCT2/files/12063357/Botany.Breakers.zip)
 Does this work?

**Comment by James103** [comment]
> [Botany Breakers.zip](https://github.com/OpenRCT2/OpenRCT2/files/12063357/Botany.Breakers.zip)
> Does this work?

Yes, I was able to reproduce this issue. Moreover, the issue isn't limited to the provided save file; I was able to reproduce it from a fresh start on OpenRCT2, v0.4.5-204-ga7881a5 (a7881a5 on develop) provided by GitHub using the following steps:
1. Start "Forest Frontiers" or any other scenario with the Classic Mini Roller Coaster unlocked at start.
2. Build a custom-designed Classic Mini Roller Coaster.
3. Set the game language to Japanese.
4. Try to rename the ride.
5. The game hangs.

**Comment by SeonAntti** [comment]
Thank you! If this issue will be fixed in the upcoming updates, it would be appreciated.

**Comment by ZehMatt** [comment]
> 

I've tried the exact same steps but I can't get it to hang

**Comment by mrmbernardi** [comment]
The cause is an infinite loop here:

https://github.com/OpenRCT2/OpenRCT2/blob/8c0a931403e87b86ee4a2bf212fc7efe70e8e5a1/src/openrct2/drawing/Drawing.String.cpp#L193

**Comment by ZehMatt** [comment]
> The cause is an infinite loop here:
> 
> https://github.com/OpenRCT2/OpenRCT2/blob/8c0a931403e87b86ee4a2bf212fc7efe70e8e5a1/src/openrct2/drawing/Drawing.String.cpp#L193

I've looked into this and its actually not simple to fix, I did clamp the codepoint iterator to never exceed the index which then properly matches the end iterator but it will just crash at GfxWrapString next, the whole utf8/unicode is quite the mess and nothing is even checking the bounds, this is a huge can of worms.

**Comment by mrmbernardi** [comment]
I did that and I was able to get something that didn't crash. However the resulting text in the box is janky and the root cause appears to be a bad string in the data. I'm also getting the same hang when building my own junior coaster in an empty park. 


Try the following change from here: https://github.com/OpenRCT2/OpenRCT2/blob/8c0a931403e87b86ee4a2bf212fc7efe70e8e5a1/src/openrct2/core/String.hpp#L217

Replace it with:
```        
        iterator& operator++()
        {
            if (_index < _str.size())
            {
                const utf8* nextch;
                GetNextCodepoint(&_str[_index], &nextch);
                _index = nextch - _str.data();
            }
            else 
            {
                _index = _str.size();
            }
            return *this;
        }
        iterator operator++(int)
        {
            auto result = *this;
            if (_index < _str.size())
            {
                const utf8* nextch;
                GetNextCodepoint(&_str[_index], &nextch);
                _index = nextch - _str.data();
            }
            else 
            {
                _index = _str.size();
            }
            return result;
        }
```
